src/Site.js

```diff
@@ -655,13 +662,23 @@ Site.prototype.regenerateAffectedPages = function (filePaths) {
           logger.error(err);
           return Promise.reject(new Error(`Error while generating ${page.sourcePath}`));
         }));
+      this.modifiedPageSrc.push(page.src);
     }
   });
 
   logger.info(`Rebuilding ${processingFiles.length} pages`);
 
   return new Promise((resolve, reject) => {
     Promise.all(processingFiles)
+      .then(() => {
+        this.pages.forEach((page) => {
+          if (this.modifiedPageSrc.includes(page.src)) {
+            page.collectHeadingsAndKeywords();
+            page.insertPageNav();
+            page.concatenateHeadingsAndKeywords();
```

Comment:
This part (and some affected parts such as `collectHeadingsAndKeywords`), is concentrated on regenerating the headings and keywords whenever it is modified during the live preview right? It is already done in #558 and #568, so changes will have to be made accordingly, so just a heads up.